Utilities/Cleanup-AzureStorageAccounts.ps1

```diff
@@ -0,0 +1,260 @@
+        .\Utilities\AddAzureRmAccountFromSecretsFile.ps1 -customSecretsFilePath $SecretFilePath
+        $SubID = ([xml](Get-Content $SecretFilePath)).Secrets.SubscriptionID
+    } else {
+        Write-LogInfo "Secret File not provided. Checking subscriptions..."
+        $Subscriptions = Get-AzSubscription
+        if ($Subscriptions) {
+            $SelectedSubscription = Select-AzSubscription -Subscription $Subscriptions[0].Id
+            Write-LogInfo "Selected $($SelectedSubscription.Name)"
+            $SubID = $Subscriptions[0].Id
+        } else {
+            Write-LogErr "Powershell Session is not authenticated / User doesn't have access to any subscription. Exiting."
+            exit 1
+        }
+    }
+    #endregion
+    Function Check-Pattern ($DiskName, $Patterns) {
+        $retValue = $false
+        if ( $Patterns ) {
+            $Patterns = $Patterns.Split(",")
+            foreach ($item in $Patterns) {
+                if ($DiskName -imatch $item) {
+                    $retValue = $true
+                    break;
+                }
+            }
+        }
+        return $retValue
+    }
+
+    Function Check-FileAge ($Disk) {
+        $DiskCreationTime = $Disk.ICloudBlob.Properties.Created.UtcDateTime
+        $CurrentTime = (Get-Date).ToUniversalTime()
+        $FileAge = ($CurrentTime - $DiskCreationTime).Days
+        return $FileAge
+    }
+
+    Function New-DiskObject ($disk) {
+        $DiskObject = New-Object -TypeName psobject
+        $DiskObject | Add-Member -MemberType NoteProperty -Name Subscription -Value $SubID
+        $DiskObject | Add-Member -MemberType NoteProperty -Name DiskName -Value $disk.Name
+        $size = [math]::Round($disk.Length / 1GB, 3)
+        $DiskObject | Add-Member -MemberType NoteProperty -Name Size -Value $size
+        $DiskObject | Add-Member -MemberType NoteProperty -Name Age -Value $null
+        $DiskObject | Add-Member -MemberType NoteProperty -Name LockStatus -Value $disk.ICloudBlob.Properties.LeaseStatus
+        $DiskObject | Add-Member -MemberType NoteProperty -Name StorageAccount -Value $disk.ICloudBlob.ServiceClient.StorageUri.PrimaryUri.Host.Split(".")[0]
+        $DiskObject | Add-Member -MemberType NoteProperty -Name ResourceGroup -Value $disk.ICloudBlob.ServiceClient.StorageUri.PrimaryUri.Host.Split(".")[0]
+        $DiskObject | Add-Member -MemberType NoteProperty -Name Region -Value $null
+        $DiskObject | Add-Member -MemberType NoteProperty -Name Created -Value $disk.ICloudBlob.Properties.Created.UtcDateTime
+        $DiskObject | Add-Member -MemberType NoteProperty -Name Modified -Value $disk.ICloudBlob.Properties.LastModified.UtcDateTime
+        $DiskObject | Add-Member -MemberType NoteProperty -Name VMName -Value $disk.ICloudBlob.Metadata["MicrosoftAzureCompute_VMName"]
+        $DiskObject | Add-Member -MemberType NoteProperty -Name VMResourceGroup -Value $disk.ICloudBlob.Metadata["MicrosoftAzureCompute_ResourceGroupName"]
```

Comment:
I didn't find this property 'MicrosoftAzureCompute_ResourceGroupName' in metadata.
```
$disk.ICloudBlob.Metadata

Key                            Value                                                   
---                            -----                                                   
MicrosoftAzureCompute_DiskId   2d896be3-33b4-43ba-89af-7a586cde09df                    
MicrosoftAzureCompute_DiskName LISAv2-OneVM-upstreamk-OC84-20190715021959-role-0-OSDisk
MicrosoftAzureCompute_DiskType OSDisk                                                  
MicrosoftAzureCompute_VMName   LISAv2-OneVM-upstreamk-OC84-20190715021959-role-0       
```

Reply:
It is present for Locked VHDs.